### ISSUE TITLE
Stored Logger issue

### ISSUE THREAD

**Issue by ghost** [falsechicken/AntiCombatLog#2]
Ok, it seems the server keeps you listed as a logger so when you come back to your gear gone then play some more and logout normally your gear is removed again at relog if server hasn't restarted yet.

**Comment by ghost** [comment]
I'm not sure if plugins can create log files in their directory but you could make a logger list so its checked on player login then if they are in it it clears inventory and removed name/id from list. Or setup a SQL db to take care of this so you have a know combat logger db for admin uses and a db to check for clear, remove gear if found, move user/id to know table and remove from checklist or use a .dat file. Been a long time since I programmed anything from college.

**Comment by falsechicken** [comment]
Yeah. I could do flatfile (a saved config) for players but performance would be a concern. Especially on large public servers where people may sign off and never sign on again. The file would get so big and reading it in at boot and writing it out at shutdown would get pretty slow. Also corruption would be a concern. If something happens during this long writing process or it gets shut off without a proper shutdown.

SQL would work but I kinda do not want to add it as a dependency. I would like everyone to be able to use the plug-in. Ill work on something after I get these first bugs ironed out. Also I am open to pull requests if anyone wants to help :).

**Comment by ghost** [comment]
Yeah I guess that is a proper concern about unwanted performance degradation. Is it possible to utilize SQL as an option usage? So players can either use as is leaving SQL enabled as false or for those who want to use it as true? I guess my main concern is false positives and redundancy. 

I have added you on steam if you wouldn't mind accepting to chat that way or easier way for me to send test results. I run on a fairly full server and if issues arise it's easier to contact that way if need be. 

I also read your code and it seems that the logger is now removed from memory to prevent the recurrence of inventory clear is that correct?

**Comment by falsechicken** [comment]
I removed a lot of the functionality to a game component. Its attached to the player and tracks itself. Its been easier for me to work with.

**Comment by falsechicken** [comment]
This has been addressed in the beta with 67894325db4f18162a6e99354fa165488dfcc656. Should be released soon once some testing goes into it. The inventory file on disk is now deleted when they disconnect. Eliminating the need to store the loggers. The only problem now is that the list of loggers is still in memory so the players logging back in after being punished will not be told of their punishment if the server has restarted since.

But this does prevent the item dupe bug.
